### PR TITLE
Added some new features

### PR DESCRIPTION
**Pull request by Sekobb**

### CODE REVIEW

**Comment by Sekobb** [comment]
Added communicate about invalid credentials
Added search bar, added logout possibility on all views

File: PZ/WebClient/tests.py [hunk 1]
```diff
@@ -1,3 +1,103 @@
 from django.test import TestCase
+from .models import *
+from .Connector import *
 
 # Create your tests here.
+url_local = 'http://localhost:8002/'
+url_resources = 'http://localhost:8002/resources/'
+url_measurements = 'http://localhost:8002/measurements/'
+
+class ConnectorTests(TestCase):
```
Comment (PatrykMatwijczuk) [inline PZ/WebClient/tests.py:10]: The tests should be independent of the machine. Do not use guid or other data from your local database.
Mocking is good practise:
http://stackoverflow.com/questions/15753390/python-mock-requests-and-the-response

File: PZ/WebClient/tests.py [hunk 2]
```diff
@@ -1,3 +1,103 @@
...
+
+    def test_get_resource(self):
+        print("Test if get_resource returns resource")
+        con = Connector(url_resources)
+        self.assertTrue(con.get_resource_id('a20eec48-b949-4cb6-ba21-86c61b6ba282'))
+
+    def test_is_get_resource_instance_of_resources(self):
+        print("Tests if get_resource returns Resources object")
+        con = Connector(url_resources)
+        self.assertIsInstance(con.get_resource_id('a20eec48-b949-4cb6-ba21-86c61b6ba282'), Resources)
```
Comment (PatrykMatwijczuk) [inline PZ/WebClient/tests.py:62]: multiple repeating the same string

**Review by PatrykMatwijczuk — CHANGES_REQUESTED** [review]

File: PZ/WebClient/views.py [hunk 3]
```diff
@@ -101,8 +101,12 @@ def hosts(request, monitor_id):
     current_monitor = Monitors.objects.get(id=monitor_id)
     c = Connector(urljoin(current_monitor.monitor_domain, 'resources'))
 
+    if request.method == 'GET':
+        search_query = request.GET.get('name', None)
+        c._payload = {"name": search_query}
```
Comment (Dorrin93) [inline PZ/WebClient/views.py:106]: `Connector` has payload as property, so you should use `.payload` instead of `._payload`

**Review by Dorrin93 — APPROVED** [review]
Nothing critical, so changes approved

File: PZ/WebClient/views.py [hunk 4]
```diff
@@ -53,15 +53,15 @@ def create_monitor(request):
             args.update(csrf(request))
             args['error'] = 'You already created monitor with selected name or domain. Choose another one.'
 
-            return render_to_response('create_monitor.html', args)
+            return render_to_response('create_monitor.html', {"full_name": request.user.username}, args)
         else:
             m = Monitors(monitor_name=name, monitor_domain=domain, user_id=request.user)
```
Comment (PatrykMatwijczuk) [inline PZ/WebClient/views.py:58]: I think that name "monitor" is better than "m"

File: PZ/templates/measurements.html [hunk 5]
```diff
@@ -1,5 +1,7 @@
 {% extends "base.html" %}
 {% block content %}
+    <p align="right"><font color="white">Logged as {{ full_name }} </font></p>
```
Comment (PatrykMatwijczuk) [inline PZ/templates/measurements.html:3]: multiple times use the same code. the better way is create separate base html file. 
http://stackoverflow.com/questions/31456576/django-footer-and-header-on-each-page-with-extends

**Pull request merged**